src/Common/Hooks/useStateFilters/useStateFilters.tsx

```diff
@@ -38,9 +38,29 @@ const useStateFilters = <T = string,>({
         sortOrder: SortingOrder.ASC,
         sortBy: initialSortKey,
     })
-
     const { sortBy, sortOrder } = sortingConfig
 
+    const [pagination, setPagination] = useState<PaginationType>({
+        pageSize: DEFAULT_BASE_PAGE_SIZE,
+        offset: 0,
```

Comment:
Imo, since this is a sister hook of useUrlFilters, we should be using pageSize and pageNumber in the state and compute the offset at runtime